curriculum/challenges/english/10-coding-interview-prep/rosetta-code/hofstadter-figure-figure-sequences.md

```diff
@@ -8,11 +8,7 @@ dashedName: hofstadter-figure-figure-sequences
 
 # --description--
 
-These two sequences of positive integers are defined as:
-
-$R(1)=1\\ ;\\ S(1)=2 \\\\R(n)=R(n-1)+S(n-1), \\quad n>1.$
-
-The sequence $S(n)$ is further defined as the sequence of positive integers not present in $R(n)$.
+You are given two sequences as `nested array`, these two sequences $R$ and $S$ start like this.
```

Comment:
I don't think we should swap the sequence definition and example, personally. That's not typical for challenges like this that I have done

Reply:
I do agree this is one of these when it click it's hard to forget. I am planning to start writing an article to help it clicking. Instead of swapping it and making it worse

Reply:
When I came back with fresher mind, I noticed that I misunderstood you. I will revert the order.

Reply:
GitHub's interface is fighting me.

What about something simpler:

```
The Hofstadter Figure-Figure sequences $R_n$ and $S_n$ are given by

$R_1 = 1\\ ;\\ S_1 = 2 \\\\R_n = R_{n-1} + S_{n-1}, \\quad n>1.$

Specifically, the sequence $R_n$ contains the values

<pre>1, 3, 7, 12, 18, ...</pre>

and the sequence $S_n$ contains the values

<pre>2, 4, 5, 6, 8, ...</pre>

The sequence $R_n$ is defined by the recurrence relation $R_n = R_{n-1} + S_{n-1}$, and $S_n$ is defined as sequence of positive integers that are not included in the sequence $R_n$.

# --instructions--

Create two functions named `ffr` and `ffs` that return `R(n)` or `S(n)`, respectively, for any index `n`. Note that the Hofgstadter Figure-Figure sequences are 1-indexed, with $R_1 = 1$ and $S_1 = 2$.

No maximum value for `n` should be assumed.

**References**

 <p>Rosetta: <a href='https://rosettacode.org/wiki/Hofstadter_Figure-Figure_sequences' target='_blank'>Hofstadter Figure-Figure sequences</a></p>.

```